File: frontend/bookstore-app/src/types/usuario.ts
export interface Usuario {
  id: number;
  username: string;
  nombreCompleto: string;
  email?: string;
  rol: string;
  activo: boolean;
  fechaCreacion: string;
  ultimoAcceso?: string;
}

export interface LoginRequest {
  username: string;
  password: string;
}

export interface LoginResponse {
  token: string;
  usuario: Usuario;
}

export interface CreateUsuarioDto {
  username: string;
  password: string;
  nombreCompleto: string;
  email?: string;
  rol: string;
}

export interface UpdateUsuarioDto {
  nombreCompleto: string;
  email?: string;
  rol: string;
  activo: boolean;
}

export interface ChangePasswordDto {
  newPassword: string;
}

File: backend/BookstoreAPI/Models/Usuario.cs
namespace BookstoreAPI.Models
{
    public class Usuario
    {
        public int Id { get; set; }
        public string Username { get; set; } = string.Empty;
        public string PasswordHash { get; set; } = string.Empty;
        public string NombreCompleto { get; set; } = string.Empty;
        public string? Email { get; set; }
        public string Rol { get; set; } = "usuario";
        public bool Activo { get; set; } = true;
        public DateTime FechaCreacion { get; set; }
        public DateTime? UltimoAcceso { get; set; }
    }
}
